float32 new_threshold
---
bool success